resin_calculator/artifact_calc.py

```diff
@@ -126,11 +134,13 @@ def select_artifact_set(rarity):
     valid_choice = False
     while valid_choice == False:
         artifact_set = user_input("What set are you looking for?")
-        if artifact_set in valid_sets:
-            valid_choice = True
+        if validate_set_selection(artifact_set, valid_sets):
+            return artifact_set
         else:
             print("Invalid Choice\n")
-    return artifact_set
+
+def validate_set_selection(artifact_set, valid_sets):
+    return artifact_set in valid_sets
```

Comment:
I think you can have this function do a little more. I would move the calculation of `valid_sets` in here. That will make the test cases a little more straightforward I think, and you won't have to copy any data.

Reply:
Makes sense, modifications made. Also made the valid sets a global variable.